File: JavaAndScala/src/main/scala/org/lang/scala/conclusion/regex/common/Captcha.scala
package org.lang.scala.conclusion.regex.common

import scala.util.Random

/**
 * 	This is a stand-alone object to randomly generate a captcha
 * 
 * 	@author VinceYuan
 */
object Captcha {
  
	/*	This is the object array to sample the number for each digit	 */
	val objArr = Array(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 'A')

	/**
	 * 	This is a method to get a captcha
	 * 	-- The length of captcha is between 6 and 9
	 * 	-- The captcha may consist of number and letter (A)
	 */
	def getCaptcha(): String = {
		
		/*	Randomly decide the number of digits for captcha: 6 - 9 digits  */
	  val numOfDigits = new Random().nextInt(4) + 6
		
		/*	Get the captcha	 */
		val stringBuilder = new StringBuilder();
		for (i <- 0 until numOfDigits) {
			stringBuilder.append(objArr(new Random().nextInt(objArr.length)))
		}
	  stringBuilder.toString()
	}
}

File: JavaAndScala/src/main/java/com/java/se/conclusion/regex/common/Captcha.java
package com.java.se.conclusion.regex.common;

import java.util.Random;

/**
 * 	This is a class to randomly generate a captcha
 * 
 * @author VinceYuan
 *
 */
public class Captcha {
	
	/*	This is the object array to sample the number for each digit	 */
	private static Object[] objectArray = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 'A'};

	/**
	 * 	This is the static method to return a captcha
	 * 	-- The length of captcha is between 6 and 9
	 * 	-- The captcha may consist of number and letter (A)
	 * @return
	 */
	public static String getCaptcha() {
		
		/*	Randomly decide the number of digits for captcha: 6 - 9 digits  */
		int numberOfDigits = (int) (Math.random()*4 + 6);
		
		/*	Get the captcha	 */
		StringBuffer stringBuffer = new StringBuffer();
		for (int i = 0; i < numberOfDigits; i++) {
			stringBuffer.append(objectArray[new Random().nextInt(objectArray.length)]);
		}
		return stringBuffer.toString();
	}
}
